packages/react-router/src/link.tsx

```diff
@@ -615,8 +620,10 @@ export function useLinkProps<
   const isActive = useRouterState({
     select: (s) => {
       // Compare path/hash for matches
-      const currentPathSplit = s.location.pathname.split('/')
-      const nextPathSplit = next.pathname.split('/')
+      const currentPathSplit = removeTrailingSlash(s.location.pathname).split(
+        '/',
+      )
+      const nextPathSplit = removeTrailingSlash(next.pathname).split('/')
```

Comment:
this is the fix: removing trailing slashes before splitting because a trailing slash results in an extra `''` entry